### PR TITLE
Login-form_CU-hh70ac

### PR DESCRIPTION
**Pull request by a-wilkes**

### CODE REVIEW

**Comment by a-wilkes** [comment]
Task linked: [CU-hh70ac Login form](https://app.clickup.com/t/hh70ac)

**Comment by lpboucher** [comment]
Task linked: [CU-hh70ac Login form](https://app.clickup.com/t/hh70ac)

File: src/app/modules/login/pages/login/login.page.ts [hunk 1]
```diff
@@ -1,15 +1,33 @@
...
+        private authService: AuthService,
+    ) {}
 
-  ngOnInit(): void {
-  }
+    ngOnInit(): void {
+    }
 
+    onSubmitLogin() {
+        const VALID_FORM_STATUS = "VALID";
```
Comment (lpboucher) [inline src/app/modules/login/pages/login/login.page.ts]: we should declare this as a type in shared/models/types

File: src/app/modules/login/pages/login/login.page.ts [hunk 2]
```diff
@@ -1,15 +1,33 @@
...
+        if (this.loginForm.status !== VALID_FORM_STATUS) {
+            return;
+        }
+
+        const input = this.loginForm.value;
```
Comment (lpboucher) [inline src/app/modules/login/pages/login/login.page.ts]: it is always best practice to access form fields in getters, e.g.:

get email() {
        return this.loginForm.get("email").value;
    }

**Review by lpboucher — CHANGES_REQUESTED** [review]
I have also added the tailwind forms plugin.

See here for some basic classes: https://tailwindcss-custom-forms.netlify.app/

File: src/app/modules/login/pages/login/login.page.ts [hunk 3]
```diff
@@ -1,15 +1,37 @@
...
+    ngOnInit(): void {
+    }
 
+    onSubmitLogin() {
+        if (this.loginForm.valid) {
+            this.authService.login(this.email, this.password);
+        }
+    }
+
+    get email(): string {
```
Comment (lpboucher) [inline src/app/modules/login/pages/login/login.page.ts:30]: getters should be placed above the constructor.

Okay for now, we can change it up later

**Review by lpboucher — APPROVED** [review]

**Pull request merged**